### DISCUSSION TITLE
S3 bucket in aws-lambda-tools-defaults.json

### DISCUSSION THREAD

**Discussion by bmanola** [aws/aws-lambda-dotnet#1645] [Q&A] [answered]
Hello Community,

I'm currently working with AWS Application (many lambdas in one project) using the AWS Toolkit for Visual Studio and have encountered a situation with my deployment process. I have a configuration set up in the aws-lambda-tools-defaults.json file, specifically referencing an S3 bucket in the s3-bucket field.  Recently, I deleted this S3 bucket ,it was empty, and am now considering recreating it with the same name.
This AWS lambda uses serverless.template

My concerns and questions are as follows:
Will deleting and potentially recreating the S3 bucket with the same name affect my already deployed Lambda functions or their versions? 
Is there any risk to the stability or accessibility of the current functions?
If I recreate the bucket with the same name, will my deployment process seamlessly continue, or are there additional considerations I need to be aware of?

Thank you in advance for your help and suggestions.

**Comment by ashishdhingra** [comment, marked as answer]
@bmanola Good morning. The Lambda function deployment process uses S3 bucket (if provided) to upload the ZIP file package containing Lambda code to the S3 bucket (else it uses MemoeryStream BLOB). The uploaded object key (this is autogenerated based on `-sp` parameter) is used in [CreateFunction](https://docs.aws.amazon.com/lambda/latest/dg/API_CreateFunction.html) request object, to point to the ZIP file in S3 bucket as function code. Once deployed, you could safely delete S3 bucket without any side-effects.

Thanks,
Ashish
